app/assets/javascripts/router.js

```diff
@@ -388,7 +400,24 @@ class ReactRouter extends React.Component<Props> {
                 }}
               />
               <Route path="/spotlight" component={SpotlightView} />
-              <Route path="/datasets/:id/view" render={this.tracingViewMode} />
+              <Route
+                path="/datasets/:organizationName/:datasetName/view"
+                render={this.tracingViewMode}
+              />
+              <Route
+                path="/datasets/:id/view"
+                render={({ match }: ContextRouter) => (
+                  <AsyncRedirect
+                    redirectTo={async () => {
+                      const datasetName = match.params.id || "";
+                      const organizationName = await getOrganizationForDataset(datasetName);
+                      return `/datasets/${organizationName}/${datasetName}/view${
+                        window.location.search
```

Comment:
Can you use `location` from `ContextRouter`? So, in line 409, it would read `({ match, location }): ...`?